src/components/shared/MarkerTooltipContents.js

```diff
@@ -240,21 +241,37 @@ function _filterInterestingPhaseTimes(
 
 function _markerBacktrace(
   marker: TracingMarker,
-  data: StyleMarkerPayload | PaintProfilerMarkerTracing | DOMEventMarkerPayload,
+  data:
+    | StyleMarkerPayload
+    | PaintProfilerMarkerTracing
+    | DOMEventMarkerPayload
+    | FrameConstructionMarkerPayload,
   thread: Thread,
   implementationFilter: ImplementationFilter
 ): React.Node {
-  if (data.category === 'DOMEvent') {
-    const latency =
-      data.timeStamp === undefined
-        ? null
-        : formatMilliseconds(marker.start - data.timeStamp);
-    return (
-      <div className="tooltipDetails">
-        {_markerDetail('type', 'Type', data.eventType)}
-        {latency === null ? null : _markerDetail('latency', 'Latency', latency)}
-      </div>
-    );
+  switch (data.category) {
+    case 'DOMEvent': {
+      const latency =
+        data.timeStamp === undefined
+          ? null
+          : formatMilliseconds(marker.start - data.timeStamp);
+      return (
+        <div className="tooltipDetails">
+          {_markerDetail('type', 'Type', data.eventType)}
+          {latency === null
+            ? null
+            : _markerDetail('latency', 'Latency', latency)}
+        </div>
+      );
+    }
+    case 'Frame Construction':
+      return (
+        <div className="tooltipDetails">
+          {_markerDetail('category', 'Category', data.category)}
+        </div>
+      );
```

Comment:
From the type, we can see there could be a "cause". But here because you're returning right away we never reach the "cause" block below. So I believe there's a mistake somewhere :)

Reply:
The same happens for the DOMEvents above actually, I wonder if that was on purpose. @mstange ?

Reply:
Gecko doesn't produce a "cause" for these marker types at the moment, so I don't know if it makes a difference.